cli/connhelper/commandconn/commandconn.go

```diff
@@ -178,7 +182,17 @@ func (c *commandConn) CloseRead() error {
 
 func (c *commandConn) Read(p []byte) (int, error) {
 	n, err := c.stdout.Read(p)
-	if err == io.EOF || errors.Is(err, os.ErrClosed) {
+	// check after the call to Read, since
+	// it is blocking, and while waiting on it
+	// Close might get called
+	if c.closing.Load() {
```

Comment:
Some observations (but may be intentional)

- `os.ErrClosed` is no longer handled in the new code
-  is it safe to discard `err` if `c.closing.Load()` is true? (not sure if there's other errors that may return)

Reply:
That code isn't from the last commit – actually comparing to the pre-this-PR code, we were never handling `os.ErrClosed()` (although we should have at the time, since this is mostly cause by us closing the connection ourselves).
It should be safe to discard errors coming from the pipe when the connection is closing, since we've already called `Close()` if there is any issue terminating the command they will be picked up there.